contracts/USM.sol

```diff
@@ -169,22 +177,27 @@ contract USM is ERC20 {
     /** INTERNAL FUNCTIONS */
 
     /**
-     * @notice Set the latest fum price. Emits a LatestFumPriceChanged event.
+     * @notice Set the min fum price, based on the current oracle price and debt ratio. Emits a MinFumBuyPriceChanged event.
      */
-    function _setLatestFumPrice() internal {
-        uint previous = latestFumPrice;
-        latestFumPrice = fumPrice();
-        emit LatestFumPriceChanged(previous, latestFumPrice);
+    function _updateMinFumBuyPrice() internal {
+        uint previous = minFumBuyPrice;
+        if (debtRatio() <= MAX_DEBT_RATIO) {                // We've dropped below (or were already below, whatev) max debt ratio
+            minFumBuyPrice = 0;                             // Clear mfbp
+        } else if (minFumBuyPrice == 0) {                   // We were < max debt ratio, but have now crossed above - so set mfbp
+            // This calculates the mfbp, in ETH terms, at the point we crossed max debt ratio.  Note how few vars it depends on:
+            minFumBuyPrice = (WAD - MAX_DEBT_RATIO).wadMul(ethPool()).wadDiv(fum.totalSupply());
```

Comment:
We should spell out all these formulas in the comments, such as this example:
```
/**
 * @dev Implements this formula:
 *           y ** z
 *   x = -----------
 *            1 - z
 * 
```

Reply:
Done - but have a look at my @<!-- -->dev comment and feel free to be picky, I'm learning here...

Reply:
That is great, I love it.